gramps/plugins/tool/check.py

```diff
@@ -2540,6 +2544,69 @@ def check_note_links(self):
+        standard_roles = set(EventRoleType().get_standard_names())
+        duplicate_roles = standard_roles.intersection(custom_roles)
+        self.duplicated_event_role_names = len(duplicate_roles)
+        if self.duplicated_event_role_names:
+            logging.info(
+                "Fixing %d custom event role names which duplicate standard role names.",
+                self.duplicated_event_role_names,
+            )
+
+            # there are some custom roles which duplicate standard roles.
+            # it is not guaranteed that the duplicate custom roles are actually used by any EventRefs
+
+            # loop over all people and coalesce duplicate custom event role names
+            self.progress.set_pass(
+                _("Coalescing duplicate roles in events referenced by people"),
+                self.db.get_number_of_people(),
+            )
+            for person in self.db.iter_people():
+                self.progress.step()
+                updated = False
+                for event_ref in person.get_event_ref_list():
+                    event_role = event_ref.get_role()
+                    # only update custom event roles which have the same name as one of the duplicate event role names
+                    if (event_role.value == EventRoleType.CUSTOM) and (
+                        event_role.string in duplicate_roles
+                    ):
+                        # reassigning the string will cause the standard value to be used
+                        event_role.string = event_role.string
+                        updated = True
+                if updated:
+                    self.db.commit_person(person, self.trans)
+
+            # loop over all families and coalesce duplicate custom event role names
+            self.progress.set_pass(
+                _("Coalescing duplicate roles in events referenced by families"),
+                self.db.get_number_of_families(),
+            )
+            for family in self.db.iter_families():
+                self.progress.step()
+                updated = False
+                for event_ref in family.get_event_ref_list():
+                    event_role = event_ref.get_role()
+                    # only update custom event roles which have the same name as one of the duplicate event role names
+                    if (event_role.value == EventRoleType.CUSTOM) and (
+                        event_role.string in duplicate_roles
+                    ):
+                        # reassigning the string will cause the standard value to be used
+                        event_role.string = event_role.string
+                        updated = True
+                if updated:
+                    self.db.commit_family(family, self.trans)
+
+            # finally delete the duplicate custom values
+            event_role_names = getattr(self.db, "event_role_names", None)
+            for role_name in duplicate_roles:
+                event_role_names.remove(role_name)
```

Comment:
I tried to use `map(event_role_names, duplicate_roles)` but it didn't work. Neither did `map(event_role_names, iter(duplicate_roles))`
If anyone knows why, I'm curious.

Reply:
What do you expect it to do?

Reply:
Thanks for the explanation @QuLogic. I'll leave the code as it is.